ibm/service/transitgateway/resource_ibm_tg_gateway_connection_prefix_filter.go

```diff
@@ -0,0 +1,288 @@
+		before := d.Get(tgBefore).(string)
+		createPrefixFilterOptions.SetBefore(before)
+	}
+	if _, ok := d.GetOk(tgGe); ok {
+		ge := int64(d.Get(tgGe).(int))
+		createPrefixFilterOptions.SetGe(ge)
+	}
+	if _, ok := d.GetOk(tgLe); ok {
+		le := int64(d.Get(tgLe).(int))
+		createPrefixFilterOptions.SetLe(le)
+	}
+
+	prefixFilter, response, err := client.CreateTransitGatewayConnectionPrefixFilter(createPrefixFilterOptions)
+	if err != nil {
+		return fmt.Errorf("[ERROR] Create Transit Gateway connection prefix filter err %s\n%s", err, response)
+	}
+	d.SetId(fmt.Sprintf("%s/%s/%s", gatewayId, connectionId, *prefixFilter.ID))
+
+	return resourceIBMTransitGatewayConnectionPrefixFilterRead(d, meta)
+}
+
+func resourceIBMTransitGatewayConnectionPrefixFilterRead(d *schema.ResourceData, meta interface{}) error {
+
+	client, err := transitgatewayClient(meta)
+	if err != nil {
+		return err
+	}
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return err
+	}
+
+	gatewayId := parts[0]
+	connectionId := parts[1]
+	filterId := parts[2]
+
+	getTransitGatewayConnectionPrefixFilterOptionsModel := &transitgatewayapisv1.GetTransitGatewayConnectionPrefixFilterOptions{}
+	getTransitGatewayConnectionPrefixFilterOptionsModel.SetTransitGatewayID(gatewayId)
+	getTransitGatewayConnectionPrefixFilterOptionsModel.SetID(connectionId)
+	getTransitGatewayConnectionPrefixFilterOptionsModel.SetFilterID(filterId)
+	prefixFilter, response, err := client.GetTransitGatewayConnectionPrefixFilter(getTransitGatewayConnectionPrefixFilterOptionsModel)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("[ERROR] Error while retrieving transit gateway connection prefix filter (%s): %s\n%s", filterId, err, response)
+	}
+
+	d.Set(tgPrefixFilterId, *prefixFilter.ID)
+	d.Set(tgCreatedAt, prefixFilter.CreatedAt.String())
+	d.Set(tgPrefix, prefixFilter.Prefix)
+
+	if prefixFilter.UpdatedAt != nil {
+		d.Set(tgUpdatedAt, prefixFilter.UpdatedAt.String())
+	}
+	if prefixFilter.Before != nil {
+		d.Set(tgBefore, prefixFilter.Before)
+	}
+	if prefixFilter.Ge != nil {
+		d.Set(tgGe, prefixFilter.Ge)
+	}
+	if prefixFilter.Le != nil {
+		d.Set(tgLe, prefixFilter.Le)
+	}
+
+	return nil
+}
+
+func resourceIBMTransitGatewayConnectionPrefixFilterUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	client, err := transitgatewayClient(meta)
+	if err != nil {
+		return err
+	}
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return err
+	}
+	gatewayId := parts[0]
+	connectionId := parts[1]
+	filterId := parts[2]
+
+	// Get the prefix filter (to compare changes)
+	getPrefixFilterOptions := &transitgatewayapisv1.GetTransitGatewayConnectionPrefixFilterOptions{}
+	getPrefixFilterOptions.SetTransitGatewayID(gatewayId)
+	getPrefixFilterOptions.SetID(connectionId)
+	getPrefixFilterOptions.SetFilterID(filterId)
+	_, response, err := client.GetTransitGatewayConnectionPrefixFilter(getPrefixFilterOptions)
```

Comment:
Response of this Get call is not used any where.. Is this call required here?

Reply:
I removed this